src/firefly/java/edu/caltech/ipac/firefly/server/packagedata/ZipHandler.java

```diff
@@ -173,9 +177,32 @@ private static InputStream getInputStream(String filename, FileInfo fi, File bas
                 URLConnection uc = URLDownload.makeConnection(url, cookies, headers);
                 uc.setRequestProperty("Accept", "text/plain");
 
+                String suggestedFilename = URLDownload.getSugestedFileName(uc);
+
+                //fallback to filename from URL path if content-disposition header is missing
+                if (isEmpty(suggestedFilename)) {
+                    String urlPath = url.getPath();
+                    suggestedFilename = urlPath.substring(urlPath.lastIndexOf('/') + 1);
+                }
+
+                //strip surrounding quotes, if any
+                suggestedFilename = suggestedFilename.replaceAll("^\"|\"$", "");
```

Comment:
These logic should only apply when `getFileNameFromUrl`.  But, it's not enough.  I would also sanitize it to remove illegal characters in file path/name.
A quick and dirty way is to do this:
```
suggestedFilename.replaceAll("[^a-zA-Z0-9._-]", "_");        // all chars not in this set will be converted to '_'
```